### ISSUE TITLE
False negative for D403 when the first sentence only has one word but the whole docstring is longer

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#13167]
[D403](https://docs.astral.sh/ruff/rules/first-line-capitalized/) has a false negative when the docstring’s first word (tokenized by spaces) is not ASCII alphabetic _and_ the docstring either contains a space or doesn’t end with ASCII terminal punctuation. This is similar to #10775. #10776 only fixed the case where a docstring contains no spaces and ends with ASCII terminal punctuation.

In the following example, `f1` has a true positive and `f2` has a false negative.

```console
$ ruff --version
ruff 0.6.3
$ cat d403.py
def f1():
    """footles."""
def f2():
    """footles.

    This is a no-op.
    """
$ ruff check -q --output-format concise --isolated --select D403 d403.py
d403.py:2:5: D403 [*] First word of the first line should be capitalized: `footles` -> `Footles`
```

**Comment by MichaReiser** [comment]
The current behavior is insofar intentional that it matches [pydocstyle](https://github.com/PyCQA/pydocstyle/blob/8d0cdfc93e86e096bb5753f07dc5c7c373e63837/src/pydocstyle/checker.py#L574-L589)

We would have to take a closer look why pydocstyle deliberately excludes all first words that contain any non ASCII characters.

**Comment by dscorbett** [comment]
Ruff’s D403 already doesn’t match pydocstyle: Ruff flags `f1` and pydocstyle doesn’t. This is intentional due to #10775. As long as Ruff’s behavior is different, I think it should be self-consistent.
